File: 1498-find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree/1498-find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */

class Solution {
public:
    TreeNode* getTargetCopy(TreeNode* original, TreeNode* cloned, TreeNode* target) {

        if (original==NULL){
            return NULL;
        }

        if (original->val == target->val){
            return cloned;
        }

        TreeNode* left = getTargetCopy(original->left, cloned->left, target);

        if (left != NULL){
            return left;
        }

        TreeNode* right = getTargetCopy(original->right, cloned->right, target);
        if (right != NULL){
            return right;
        }

        return NULL;
        
    }
};

File: 1498-find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree/1498-find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def getTargetCopy(self, original: TreeNode, cloned: TreeNode, target: TreeNode) -> TreeNode:
        if not original:
            return None
        
        if original.val == target.val:
            return cloned
        
        left = self.getTargetCopy(original.left, cloned.left, target)
        if left:
            return left
        right = self.getTargetCopy(original.right, cloned.right, target)
        if right:
            return right

        return None
        